modules/tracktion_engine/playback/graph/tracktion_LoopingMidiNode.cpp

```diff
@@ -723,32 +727,29 @@ struct EventGenerator   : public MidiGenerator
                                             useMPEChannelMode, midiSourceID,
                                             controllerMessagesScratchBuffer);
 
-        // This isn't quite right as there could be notes that are turned on in the original buffer after the scratch buffer?
         for (const auto& e : scratchBuffer)
         {
             if (e.isNoteOn())
             {
-                if (! activeNoteList.isNoteActive (e.getChannel(), e.getNoteNumber()))
+                if (!originalState.isNoteActive(e.getChannel(), e.getNoteNumber()))
                 {
-                    cleanedBufferToMerge.add (e);
-                    activeNoteList.startNote (e.getChannel(), e.getNoteNumber());
+                    destBuffer.addMidiMessage(e, midiSourceID);
+                    activeNoteList.startNote(e.getChannel(), e.getNoteNumber());
```

Comment:
I know it's a small thing but can you add the spaced beck after `! ` and before ` (` please?
I think that will just simplify the diff a bit as there's a few lines that are just missing this space.